src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java

```diff
@@ -94,6 +94,21 @@ public static UrlRewriter getDownloaderUrlRewriter(String configPath, Reporter r
       return new UrlRewriter(log, "", new StringReader(""));
     }
 
+    // If the `configPath` is absolute, use that. Otherwise, prepend the `workspaceRoot`.
+    // There have been reports (eg. https://github.com/bazelbuild/bazel/issues/22104) that
+    // there are occasional errors when `configFile` can't be found, and when this happens
+    // investigation suggests that the current working directory isn't the workspace root.
+    Path actualConfigPath;
```

Comment:
this segment could simply be replaced with `workspaceRoot.getRelative(configPath)` -- it deals with an absolute `configPath` automatically.

Reply:
I did not know that. Thank you! Fixed